commit 40fb87129049ec5876dabf4a4d4aed6642b31f1a
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Wed Aug 17 10:44:05 2022 +0200

    wifi: mac80211: fix use-after-free
    
    We've already freed the assoc_data at this point, so need
    to use another copy of the AP (MLD) address instead.
    
    Fixes: 81151ce462e5 ("wifi: mac80211: support MLO authentication/association with one link")
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 3d4ab711f0d1..04d35cd39889 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -5124,7 +5124,7 @@ static void ieee80211_rx_mgmt_assoc_resp(struct ieee80211_sub_if_data *sdata,
 	resp.req_ies = ifmgd->assoc_req_ies;
 	resp.req_ies_len = ifmgd->assoc_req_ies_len;
 	if (sdata->vif.valid_links)
-		resp.ap_mld_addr = assoc_data->ap_addr;
+		resp.ap_mld_addr = sdata->vif.cfg.ap_addr;
 	cfg80211_rx_assoc_resp(sdata->dev, &resp);
 notify_driver:
 	drv_mgd_complete_tx(sdata->local, sdata, &info);